hiero-observability/hiero-metrics-core/docs/metrics_details.md

```diff
@@ -0,0 +1,89 @@
+| [DoubleCounter](../src/main/java/org/hiero/metrics/api/DoubleCounter.java)         | Counter  | Allows to increment `double` (only increasing).                                                                                    |
+| [StatelessMetric](../src/main/java/org/hiero/metrics/api/StatelessMetric.java)     | Gauge    | Requires zero ore more `DoubleSupplier`/`LongSupplier` (with optional labels), which will be called on export.                     |
+| [BooleanGauge](../src/main/java/org/hiero/metrics/api/BooleanGauge.java)           | Gauge    | Allows to set single `boolean` value.                                                                                              |
+| [LongGauge](../src/main/java/org/hiero/metrics/api/LongGauge.java)                 | Gauge    | Gauge to observe and store `long`. May have an accumulator applied to previous and observed values.                                |
+| [DoubleGauge](../src/main/java/org/hiero/metrics/api/DoubleGauge.java)             | Gauge    | Gauge to observe and store `double`. May have an accumulator applied to previous and observed values.                              |
+| [GenericGauge](../src/main/java/org/hiero/metrics/api/GenericGauge.java)           | Gauge    | Parametrized generic gauge that allows to observe any non-primitive type and has a converter to `double` or `long` for exporting.  |
+| [GaugeAdapter](../src/main/java/org/hiero/metrics/api/GaugeAdapter.java)           | Gauge    | A gauge to adapt to any external class that is used to observe and hold a single value. Parametrized with external datapoint type. |
+| [StatsGaugeAdapter](../src/main/java/org/hiero/metrics/api/StatsGaugeAdapter.java) | Gauge    | Similar to GaugeAdapter, but for multiple numeric values - usually different stats on same observation.                            |
+| [StateSet](../src/main/java/org/hiero/metrics/api/StateSet.java)                   | StateSet | Allows for enabling/disabling states defined by enum type.                                                                         |
+
+There is also package `org.hiero.metrics.api.stat` that contains custom statistical metrics build on top of the above metrics.
+
+### Metric Registry
+
+[MetricRegistry](../src/main/java/org/hiero/metrics/api/core/MetricRegistry.java) can be used to register new metrics and retrieve existing ones.
+Metrics Registry must have non-blank name and may have immutable list of global labels, which will be applied to all registered metrics.
+Registry can also be managed by [MetricsExportManager](../src/main/java/org/hiero/metrics/api/export/MetricsExportManager.java)
+to export all its registered metrics - see more in [Metrics Exporting](metrics_exporting.md) documentation.
+
+There are two ways to register metrics in the registry:
+- Programmatically - by using metric builders and calling `register` method of the registry.
```

Comment:
Can a single metric be registered in multiple registries? I guess, not, but would like to get it confirmed.

This is not a concern, when metrics registration SPI is used (for every registry, a metrics provider is called to register all its metrics in that registry), but it can be possible when metrics are registered explicitly